browser/src/core/Socket.js

```diff
@@ -980,7 +980,7 @@ app.definitions.Socket = L.Class.extend({
 				msg += fontsMissingObj.fontsmissing[i];
 			}
 
-			if (!this._map.welcome.isGuest() && this._map.welcome.shouldWelcome() && window.autoShowWelcome)
+			if (this._map.welcome && !this._map.welcome.isGuest() && this._map.welcome.shouldWelcome() && window.autoShowWelcome)
```

Comment:
I'm confused. This only checked that `welcome` is defined. But that wouldn't change the behavior that the "fontsmissing" dialog will *still* only work if welcome dialog is enabled, or I missed something?

Was this supposed to be  `if (!this._map.welcome || (!this._map.welcome.isGuest() && this._map.welcome.shouldWelcome() && window.autoShowWelcome))` ?

Reply:
If you check the surrounding code, you will see that both paths show the missing fonts dialog, just in a different way, see below the following line:
https://github.com/CollaboraOnline/online/blob/7d2f7360843b82908dd6c405b6815d6ea38d5563/browser/src/core/Socket.js#L983